### PR TITLE
tripDiary: add visited place cancel/delete/save buttons in Evolution

### LINKED ISSUE #1484: tripDiary: port section buttons to Evolution
See od_nationale_quebec. The following buttons should be ported to Evolution

- [ ] Save visited place (save callback is quite long)
- [ ] Cancel edit visited place
- [ ] Delete visited place
- [ ] Confirm and goto next section

### PR DESCRIPTION
**Pull request by tahini**
fixes #1484 

Add a `ButtonsVisitedPlaceConfigFactory` which adds the 3 buttons, copied from od_nationale_quebec. The cancel and delete differ mostly by the color and text, their action is identical and calls the `deleteVisitedPlace` helper function.

The save button does a lot more: it updates previous visited places to set their departure/arrival time if those questions were answered in the current place. It adds, if necessary, a previous place (in case of work on the road not starting at previous place) and a next place if it does not exist yet, with some of the fields pre-filled if possible (like home or work activities.)

## Summary by CodeRabbit

* **Breaking Changes**
  * Widget date-format option now accepts an additional optional parameter.

* **New Features**
  * Richer visited-places controls: Save, Delete, Cancel buttons and a confirm-and-continue option in demos.
  * Improved insertion flow with explicit insert helpers, empty-place placeholders, and safer neighbor/time handling.

* **Tests**
  * Expanded test coverage for visited-places buttons, insertion scenarios, and related edge cases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - `getFirstIncompleteVisitedPlace` (commit [600bb6e](https://github.com/chairemobilite/evolution/commit/600bb6ea2f019dc07400440f84c380d1f4cc1dce))
   - `reconcileVisitedPlaces` (commit [cb1e699](https://github.com/chairemobilite/evolution/commit/cb1e699ba27534c8056597045012e5e7af2157e9))
   - `deleteVisitedPlace` (commit [fd6e33b](https://github.com/chairemobilite/evolution/commit/fd6e33bfa36c16b741469b30dd5ca90ef1948083))
-  - `addVisitedPlace` (commit [3b3be2d](https://github.com/chairemobilite/evolution/commit/3b3be2d7d0d79e5f5b4928f6784e892d67562687))
+  - `insertEmptyVisitedPlace`
```
Comment (samuel-duhaime) [inline CHANGELOG.md]: No commit # for this one?

File: packages/evolution-common/src/services/odSurvey/__tests__/helpers.test.ts [hunk 2]
```diff
@@ -2262,6 +2262,59 @@ describe('insertEmptyVisitedPlace', () => {
...
 
+    each([
+        {
+            title: 'inserts an empty visited place and return the updated interview',
+            insertSequence: 2,
+            newVisitedPlace: undefined
+        },
+        {
+            title: 'insert a visited place with data and return the updated interview',
+            insertSequence: -1,
```
Comment (samuel-duhaime) [inline packages/evolution-common/src/services/odSurvey/__tests__/helpers.test.ts:2273]: Petite question. Ça fait quoi une sequence négative ?

File: packages/evolution-common/src/services/odSurvey/helpers.ts [hunk 3]
```diff
@@ -1335,18 +1335,58 @@ export const reconcileVisitedPlaces = ({
...
+/**
+ * Add a visited place at the right sequence in the visited places array. It
+ * adds the new place by calling the startAddGroupedObjects callback function
+ * and returns the updated interview, with the new place in the array. It does
+ * not modify any other property of the place or the interview.
+ *
+ * @param {Object} options - The options object.
+ * @param {Person} options.person The person the visited place belongs to
+ * @param {Journey} options.journey The journey the visited place is part of
+ * @param {Partial<VisitedPlace>} options.newVisitedPlace The new visited place
```
Comment (samuel-duhaime) [inline packages/evolution-common/src/services/odSurvey/helpers.ts]: Should we say that this parameter is optional?

**Review by samuel-duhaime — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tahini) [reply CHANGELOG.md]: The rebase and merge will assign the commit hash, we don't know it yet, we'll need to update the CHANGELOG for those cases

File: packages/evolution-common/src/services/odSurvey/__tests__/helpers.test.ts (hunk 2 above)
Comment (tahini) [reply packages/evolution-common/src/services/odSurvey/__tests__/helpers.test.ts:2273]: ça insère à la fin

**Review by kaligrafy — APPROVED** [review]
nice
